src/sysupdate/sysupdate.c

```diff
@@ -956,8 +1007,37 @@ static int verb_list(int argc, char **argv, void *userdata) {
 
         if (version)
                 return context_show_version(context, version);
-        else
+        else if (FLAGS_SET(arg_json_format_flags, SD_JSON_FORMAT_OFF))
                 return context_show_table(context);
+        else {
+                _cleanup_(sd_json_variant_unrefp) sd_json_variant *json = NULL;
+                _cleanup_strv_free_ char **versions = NULL;
+                const char *current = NULL;
+
+                for (size_t i = 0; i < context->n_update_sets; i++) {
+                        UpdateSet *us = context->update_sets[i];
+
+                        if (FLAGS_SET(us->flags, UPDATE_INSTALLED) &&
+                            FLAGS_SET(us->flags, UPDATE_NEWEST))
+                                current = us->version;
+
+                        r = strv_extend(&versions, us->version);
+                        if (r < 0)
+                                return log_oom();
+                }
+
+                r = sd_json_buildo(&json, SD_JSON_BUILD_PAIR_STRING("current", current),
+                                          SD_JSON_BUILD_PAIR_STRV("all", versions),
+                                          SD_JSON_BUILD_PAIR_STRV("appstream-url", context->appstream));
```

Comment:
hmm, you introduce that field only in a later commit in this PR, no?

please make sure that each of your commits builds cleanly, i.e. I recommend running `git fetch origin && git rebase origin/main -x "ninja -C build"` before each push to rebase your branch on git main, and then automatically build every single commit of yours. I'd guess this would fail with this PR right now, no?